control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1804,6 +1808,24 @@ func (r *reconciler) ensureIngressControllersRemoved(ctx context.Context, hcp *h
 		log.Info("There are no ingresscontrollers, nothing to do")
 		return true, nil
 	}
+
+	// If the DeletionTimestamp reaches the gracePeriod, the controller
+	// should ignore the object and continue with the deletion
+	counter := 0
+	for i := range ingressControllers.Items {
+		ic := &ingressControllers.Items[i]
+		if ic.DeletionTimestamp != nil {
+			deletionTime := ic.ObjectMeta.DeletionTimestamp.Add(DeletionSecondsTimeout)
+			if ic.ObjectMeta.DeletionTimestamp.After(deletionTime) {
```

Comment:
I don't think this is correct.
you can just use `if time.Since(ic.ObjectMeta.DeletionTimestamp) > DeletionSecondsTimeout`